back/engines/commercial/machine_translations/app/services/machine_translations/machine_translation_service.rb

```diff
@@ -1,47 +1,66 @@
 module MachineTranslations
   class MachineTranslationService
-
     def build_translation_for translatable:, attribute_name:, locale_to:
       multiloc = translatable[attribute_name]
       locale_from = multiloc_to_locale_from multiloc
       text_or_html = multiloc[locale_from]
       MachineTranslation.new(
         translatable: translatable,
-        attribute_name: attribute_name, 
-        locale_to: locale_to, 
+        attribute_name: attribute_name,
+        locale_to: locale_to,
         translation: translate(text_or_html, locale_from, locale_to)
-        )
+      )
     end
 
-    def assign_new_translation translation
+    def assign_new_translation(translation)
       multiloc = translation.translatable[translation.attribute_name]
       locale_from = multiloc_to_locale_from multiloc
       text_or_html = multiloc[locale_from]
       translation.assign_attributes translation: translate(text_or_html, locale_from, translation.locale_to)
     end
 
-    def multiloc_to_locale_from multiloc
+    def multiloc_to_locale_from(multiloc)
       # It would be nice if we could figure out
       # the better locale to translate from.
       multiloc.keys.first
     end
 
-    def translate text_or_html, locale_from, locale_to, retries: 10
+    def translate(text_or_html, locale_from, locale_to, retries: 10)
       # Uses Google translate
       from = locale_from[0...2]
       to = locale_to[0...2]
-      if from == to
-        text_or_html
-      else
-        while retries > 0
-          begin
-            return EasyTranslate.translate(text_or_html, :from => from, :to => to)
-          rescue EasyTranslate::EasyTranslateException => e
-            retries -= 1
-            sleep rand(60)
-          end
+
+      return text_or_html if from == to
+
+      translation = nil
+      exception = nil
+      while retries > 0
+        begin
+          translation = EasyTranslate.translate text_or_html, from: from, to: to
+          break
+        rescue EasyTranslate::EasyTranslateException => e
+          exception = e
+          retries -= 1
+          sleep rand(60)
```

Comment:
From where is this method invoked? If it is invoked from a controller, then sleeping max 10 * 59 seconds is not a good idea. Why does this method have to retry?

Reply:
It's also used when copying projects from e.g. an NL platform to an FR platform, where the content automatically gets translated.

Reply:
Yes, I changed the code so that no retries are performed through the API request, I think it's a very good remark. We don't have an explicit limit, and today we do have endpoints that are extremely slow (sometimes a minute long). This is something that will be tackled in the tech dept epic.